controller/src/main/java/org/jboss/as/controller/capability/registry/ImmutableCapabilityRegistry.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2015, Red Hat, Inc., and individual contributors as indicated
+ * by the @authors tag.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jboss.as.controller.capability.registry;
+
+import java.util.Set;
+
+import org.jboss.as.controller.CapabilityRegistry;
+import org.jboss.msc.service.ServiceName;
+
+/**
+ * A read-only view of {@link CapabilityRegistry}
+ *
+ * @author Tomaz Cerar (c) 2015 Red Hat Inc.
+ */
+public interface ImmutableCapabilityRegistry {
+    /**
+     * Gets whether a capability with the given name is registered.
```

Comment:
This can come after merge, but the javadoc in this interface could use clarifiication re: the "possible capabilities" vs the "actual" capabilities. Perhaps an explanation in the class level javadoc of the distinction between the two concepts, and then in each method a clear statement re: which of the two is relevant.